source/com.android.installreferrer/installreferrer/Transforms/Metadata.Namespaces.xml

```diff
@@ -0,0 +1,27 @@
+<metadata>
```

Comment:
Am I not allowed to change namespaces "old way" (with metadata)?

Reply:
In general, the new way is less work.  It would require no work in most new cases.  This particular one requires 2 lines the way as opposed to 4 lines the old way.

The new way also runs faster in `generator`, making builds faster.

Reply:
Now, just for changing namespaces we have 2 ways of doing it

1. XPath/XML
2. MSBuild

In a spirit of "growth mindset" let us further grow number of options to fix namespaces with:

1. Roslyn analyzers
2. Source generators

Just to make bindings more approachable for users and technology acceptance. Bindings are trivial and everybody does it just for fun and users are thrilled with alternatives. The more options the better and users will benefit by being forced to learn, just they don't know that. Options never raise questions and express freedom and inclusiveness.

Pro argument for MsBuild approach I recently heard: "new (MSBuild )way is centrally managed".

We have central metadata files in both AX and GPS-FB-MLKit which could be used for central management of namespaces.

Now, performance. I accept that MsBuild approach might be faster than XPath, but we have other problems, like hundreds if not thousands of `matched no node` which slow down builds, etc etc.

For the record:

https://github.com/xamarin/java.interop/issues/727#issuecomment-975172468

Reply:
As you said here: https://github.com/xamarin/GooglePlayServicesComponents/pull/636#discussion_r895888396:

> Options do not hurt.

😉

Reply:
A little bit if the PR is not approved for using "other way" or "old way".